apps/sms/test/unit/contacts_test.js

```diff
@@ -39,156 +39,108 @@ suite('Contacts', function(done) {
-                    ]);
-                  }
-
-                  if (filter.filterValue === 'do') {
-                    return MockContact.list([
-                      // true
-                      { givenName: ['Jane'], familyName: ['Doozer'] },
-                      // false
-                      { givenName: ['doug'], familyName: ['dooley'] },
-                      // true
-                      { givenName: ['jane'], familyName: ['doe'] },
-                      // false
-                      { givenName: ['jerry'], familyName: ['doe'] },
-                      // true
-                      { givenName: ['j'], familyName: ['dow'] },
-                      // true
-                      { givenName: ['john'], familyName: ['doland'] }
-                    ]);
-                  }
-
-                  if (filter.filterValue === 'doozer') {
-                    return MockContact.list([
-                      { givenName: ['Jane'], familyName: ['Doozer'] }
-                    ]);
-                  }
-
-                  if (filter.filterValue === 'mary') {
-                    return MockContact.list([
-                      { givenName: ['Mary Anne'], familyName: ['Jones'] }
-                    ]);
-                  }
-
-                  // Fake error
-                  if (filter.filterValue === 'callonerror') {
-                    return null;
-                  }
-
-                  // All other cases
-                  return MockContact.list();
-                }());
-
-                if (this.result === null) {
-                  this.error = {
-                    name: 'Mock missing filter params'
-                  };
-                  if (onerror) {
-                    setTimeout(function() {
-                      onerror.call(this);
-                      onerror = null;
-                    }.bind(this), 0);
-                  }
-                } else {
-                  setTimeout(function() {
-                    onsuccess.call(this);
-                    onsuccess = null;
-                  }.bind(this), 0);
-                }
-              }
-            }
-          },
-
-          onerror: {
-            set: function(callback) {
-              onerror = callback;
-              if (callback !== null) {
-                if (this.result === null && this.error !== null) {
-                  setTimeout(function() {
-                    onerror.call(this);
-                    onerror = null;
-                  }.bind(this), 0);
-                }
-              }
-            }
-          }
-        });
+        // default value
+        var result = MockContact.list();
+
+        // Supports the error case
+        if (filter == null ||
+            (!filter.filterOp || !filter.filterValue)) {
+          result = null;
+        }
+
+        if (filter.filterValue === targetLocalNumber ||
+            filter.filterValue === targetLocalEmail) {
+          result = [{
```

Comment:
actually, we should just use sinon from the test themselves rather than this :p

I'll file a follow up for this too.